tensorflow/stream_executor/stream.h

```diff
@@ -444,6 +455,30 @@ class Stream {
       const dnn::AlgorithmConfig &algorithm_config,
       dnn::ProfileResult *output_profile_result);
 
+  Stream &ThenConvolveBackwardDataWithAlgorithm(
+      const dnn::FilterDescriptor &filter_descriptor,
+      const DeviceMemory<int8> &filter_data,
+      const dnn::BatchDescriptor &output_descriptor,
+      DeviceMemory<int8> backward_output_data,
+      const dnn::ConvolutionDescriptor &convolution_descriptor,
+      const dnn::BatchDescriptor &input_descriptor,
+      DeviceMemory<int8> *backward_input_data,
+      ScratchAllocator *scratch_allocator,
+      const dnn::AlgorithmConfig &algorithm_config,
+      dnn::ProfileResult *output_profile_result) { CHECK(false); }
```

Comment:
What's the purpose of "implementing" a non-supported convolution?

Reply:
This implementation is for ThenConvolveBackwardDataWithAlgorithm.  Back propagation with integer type is not supported.

However, after I added S8 to RunCudnnConv, RunCudnnConv now calls RunCudnnConvImpl<int8, float>, which calls/requires ThenConvolveWithAlgorithm<int8> (desired) and ThenConvolveBackwardDataWithAlgorithm<int8> (undesired).  

This is the smallest change I can think of to call ThenConvolveWithAlgorithm<int8>.  Otherwise, we may have to refactor RunCudnnConvImpl and RunCudnnConv to do proper double dispatch on data type and convolution type.

Reply:
I have refactored these code to use template to dispatch for data types and each data type specialization only calls supported convolutions.  We don't need to implement such non-supported convolutions any more.